libraries/AP_HAL_ChibiOS/CANSerialRouter.cpp

```diff
@@ -117,12 +117,14 @@ void SLCANRouter::slcan2can_router_trampoline(void)
         }
         chSysUnlock();
         _slcan_if.reader();
-        if (_can_tx_queue.available() && _can_if) {
+        while (_can_tx_queue.available() && _can_if) {
             _can_tx_queue.peek(it);
             if (_can_if->send(it.frame, uavcan::MonotonicTime::fromUSec(AP_HAL::micros64() + 1000), 0)) {
```

Comment:
Should these loops break if the send fails?

Reply:
Yeah, It's probably better for them to just break on fail. I will add that.

Reply:
This is resolved now.